documentation/USAGE.rst

```diff
@@ -535,68 +535,223 @@ Further demonstrations of spatially varying boundary condition can be found
+                 {-a \left(\hat{n}\cdot\vec{d}_{fP}\right)_{f_0} + b} A_{f_0}
+   \end{aligned}
+
+Applying internal "boundary" conditions
+=======================================
 
 Applying internal boundary conditions can be achieved through the use
-of implicit and explicit sources. An equation of the form
+of implicit and explicit sources. 
+
+Internal fixed value
+--------------------
+
+An equation of the form
 
 >>> eqn = TransientTerm() == DiffusionTerm()
 
 can be constrained to have a fixed internal ``value`` at a position
 given by ``mask`` with the following alterations
 
->>> eqn = TransientTerm() == DiffusionTerm() - ImplicitSourceTerm(mask * largeValue) + mask * largeValue * value
+>>> eqn = (TransientTerm() == DiffusionTerm() 
+...                           - ImplicitSourceTerm(mask * largeValue) 
+...                           + mask * largeValue * value)
 
 The parameter ``largeValue`` must be chosen to be large enough to
 completely dominate the matrix diagonal and the RHS vector in cells
 that are masked. The ``mask`` variable would typically be a
 ``CellVariable`` boolean constructed using the cell center values.
 
-One must be careful to distinguish between constraining internal cell
-values during the solve step and simply applying arbitrary constraints
-to a ``CellVariable``. Applying a constraint,
+Internal fixed gradient
+-----------------------
+
+An equation of the form
+
+>>> eqn = TransientTerm() == DiffusionTerm(coeff=Gamma0)
+
+can be constrained to have a fixed internal ``gradient`` magnitude 
+at a position given by ``mask`` with the following alterations
+
+>>> Gamma = FaceVariable(mesh=mesh, value=Gamma0)
+>>> Gamma[mask.value] = 0.
+>>> eqn = (TransientTerm() == DiffusionTerm(coeff=Gamma) 
+...        + DiffusionTerm(coeff=largeValue * mask)
+...        - ImplicitSourceTerm(mask * largeValue * gradient 
+...                             * mesh.faceNormals).divergence)
+
+The parameter ``largeValue`` must be chosen to be large enough to
+completely dominate the matrix diagonal and the RHS vector in cells
+that are masked. The ``mask`` variable would typically be a
+``FaceVariable`` boolean constructed using the face center values.
+
+Internal Robin condition
+------------------------
+
+Nothing different needs to be done when 
+`applying Robin boundary conditions`_ at internal faces.
+
+.. note::
+
+   While we believe these derivations are "correct", they often do not 
+   seem to produce the intuitive result. At this point, we think this has 
+   to do with the pathology of "internal" boundary conditions, but remain 
+   open to other explanations. :term:`FiPy` was designed with diffuse 
+   interface treatments (phase field and level set) in mind and, as such, 
+   internal "boundaries" do not come up in our own work and have not 
+   received much attention.
+
+.. note::
+
+  One must be careful to distinguish between constraining internal cell
```

Comment:
Maybe start off with what the note is actually about and/or make it a warning.

Warning: the constraints mechanism are not designed to constrain internal values for variables that are being solved by equations. In particular, one must be careful

Reply:
agreed